### PR TITLE
roachprod-microbench: fix empty sheet issue

### PR DESCRIPTION
**Pull request by herkolategan**
This PR adds a fix for an error that occurred during sheet generation, as well as not generating empty sheets in certain scenarios. Additionally better error logging has been added to assist with any issues in the future.

Epic: None
Release Note: None

### CODE REVIEW

File: pkg/cmd/roachprod-microbench/google/service.go [hunk 1]
```diff
@@ -227,6 +230,7 @@ func (srv *Service) createRawSheet(
 	cf := condFormatting(sheetID, info.deltaCol, metric.Better < 0)
 
 	// Grid properties.
+
```
Comment (DarrylWong) [inline pkg/cmd/roachprod-microbench/google/service.go:233]: nit: accidental enter?

**Review by DarrylWong — APPROVED** [review]
LGTM!

**Review by herkolategan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113928)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong and @renatolabs)

___
*[`pkg/cmd/roachprod-microbench/google/service.go` line 233 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/113928#gh-1386856802:-NikK2yi8umLl7SrOpFF:b-jo0fpp) ([raw file](https://github.com/cockroachdb/cockroach/blob/1ee685fe14b5b14dc8f050f4048327872e9c546f/pkg/cmd/roachprod-microbench/google/service.go#L233)):*
<details><summary><i>Previously, DarrylWong wrote…</i></summary><blockquote>

nit: accidental enter?
</blockquote></details>

Hah yeah, slipped in. Thanks!

**Comment by herkolategan** [comment]
TFTR!

bors r=DarrylWong

**Comment by herkolategan** [comment]
bors retry

**Pull request merged**